src/Common/ForkManager/ForkManager.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\Common\ForkManager;
+
+use Gaming\Common\ForkManager\Exception\ForkManagerException;
+
+final class ForkManager
+{
+    /**
+     * @var Process[]
+     */
+    private array $forks;
+
+    public function __construct()
+    {
+        $this->forks = [];
+    }
+
+    /**
+     * @throws ForkManagerException
+     */
+    public function fork(Task $task): Process
+    {
+        $streamPair = StreamPair::create();
+
+        $parentPid = getmypid();
+        if ($parentPid === false) {
+            throw new ForkManagerException('Unable to get the process id.');
+        }
+
+        $forkPid = pcntl_fork();
+
+        return match ($forkPid) {
+            -1 => throw new ForkManagerException('Unable to fork.'),
+            0 => $this->runTaskAndExit($parentPid, $task, $streamPair),
+            default => $this->registerFork($forkPid, $streamPair)
+        };
+    }
+
+    public function wait(): void
```

Comment:
This function should enable a graceful shutdown. It should also tell the user which process has been terminated.

Reply:
There's now a separat signal api which allows that.